main.ts

```diff
@@ -1,135 +1,110 @@
-				.expandAndWrap("_", "_", editor),
-		});
-		this.addCommand({
-			id: "smarter-comments",
-			name: "Smarter Comments",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("%%", "%%", editor),
-		});
-		this.addCommand({
-			id: "smarter-inline-code",
-			name: "Smarter Inline Code",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("`", "`", editor),
-		});
-		this.addCommand({
-			id: "smarter-highlight",
-			name: "Smarter Highlight",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("==", "==", editor),
-		});
-		this.addCommand({
-			id: "smarter-strikethrough",
-			name: "Smarter Strikethrough",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("~~", "~~", editor),
-		});
-		this.addCommand({
-			id: "smarter-wikilink",
-			name: "Smarter wikilink (internal link)",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("[[", "]]", editor),
-		});
-		this.addCommand({
-			id: "smarter-md-link",
-			name: "Smarter Markdown Link",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("[", "]()", editor),
+		COMMANDS.forEach((command) => {
+			const { id, name, before, after } = command;
+			this.addCommand({
+				id,
+				name,
+				editorCallback: (editor) =>
+					this.expandAndWrap(before, after, editor),
+			});
 		});
 
-		console.log ("Smarter MD Hotkeys loaded.");
+		console.log("Smarter MD Hotkeys loaded.");
 	}
 
 	async onunload() {
-		console.log ("Smarter MD Hotkeys unloaded.");
+		console.log("Smarter MD Hotkeys unloaded.");
 	}
 
 	expandAndWrap(beforeStr: string, afterStr: string, editor: Editor): void {
+		// Arrow functions are simpler
+		// No need to specify the return-type, Typescript can infer it
+		const offToPos = (offset: number) => editor.offsetToPos(offset);
 
-		function offToPos(offset: number): EditorPosition {
-			return editor.offsetToPos(offset);
-		}
-		function WordUnderCursor (ep: EditorPosition): Range{
+		// Return type is infered
+		function WordUnderCursor(ep: EditorPosition) {
 			// https://codemirror.net/doc/manual.html#api_selection
-			if (editor.cm?.findWordAt) return editor.cm.findWordAt(ep); //CM5
-			else if (editor.cm?.state.wordAt) return editor.cm.state.wordAt(editor.posToOffset(ep)); //CM6
+			if (editor.cm?.findWordAt) return editor.cm.findWordAt(ep);
+			//CM5
+			else if (editor.cm?.state.wordAt)
+				return editor.cm.state.wordAt(editor.posToOffset(ep)); //CM6
 		}
-		function trimSelection (trimBefArray: string[], trimAftArray: string[]): void{
+
+		function trimSelection(
+			trimBefArray: string[],
+			trimAftArray: string[]
+		): void {
 			let selection = editor.getSelection();
 			let sp = editor.posToOffset(editor.getCursor("from"));
 
 			//before
 			let trimFinished = false;
-			while (!trimFinished){
+			while (!trimFinished) {

```

Comment:
Alot of changes are just my formatter. Spaces and brackets, usually

Reply:
I am sorry, but I think I'll undo have of them later on again, since I am used to spacing brackets my way so much :P

Reply:
Haha to each their own 😋 
I just didn't want to manually change it back